lib/galaxy/datatypes/molecules.py

```diff
@@ -101,17 +105,27 @@ def sniff(self, filename):
         >>> SDF().sniff(fname)
         False
         """
-        counter = count_special_lines("^M\s*END", filename) + count_special_lines("^\$\$\$\$", filename)
-        if counter > 0 and counter % 2 == 0:
-            return True
-        else:
-            return False
+        m_end_found = False
+        limit = 500
```

Comment:
This probably needs to be increased. I have seen molecules with 1000 atoms, so maybe 1500?

Reply:
Good point! I've increased this to 10000, but also added a strict check on the 4th line which would stop the method early for most files.